### PR TITLE
UI: Update behavior when deleting nested secret from list

### PR DESCRIPTION
**Pull request by hashishaw**
**TL;DR**
This PR removes the `withNavToNearestAncestor` mixin and updates the UX behavior in nested secret lists that are not KV version 2 (eg. KV version 1, cubbyhole). 

**Previous Behavior**
Say there's a secret engine with a secret at a deeply nested path (eg. `kv1/one/two/three/four`). A user viewing the list at `kv1/one/two/three/` can see `four` in the list, and can delete from the list view. In the previous behavior, the UI would then try to attempt to navigate to each segment ancestor above until it got to a successful response, or got back to the list-root. 

before-kv-nav

**New Behavior**
After updating everything to use `router.transitionTo`, we could no longer catch the transition error from where it was called in the mixin. Instead, we are matching the updated pattern in KV version 2, and showing an empty state message while allowing the user to navigate back to the root themselves OR adjust the filter bar to move back up the tree. This behavior is more deterministic, and simplifies the error logic slightly. 

after-kv-nav

### CODE REVIEW

**Review by Monkeychip — APPROVED** [review]
really nice solution and write up. thanks for tackling!
